src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -425,6 +440,11 @@ void Terminal::_WriteBuffer(const std::wstring_view& stringView)
             }
         }
 
+        if (proposedCursorPosition.X > bufferSize.RightInclusive())

```

Comment:
Did you intentionally put this back? It looks like @DHowett-MSFT *just* removed this in #3212

Reply:
The line the caused regression was `proposedCursorPosition.Y++;`. I only took the first line and reset `position.X` to a legal value.

Reply:
OK.

Reply:
This unfortunately caused #3277.